ui-spacetimechart/src/stories/horizontal-zoom.stories.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useEffect, useState } from 'react';
+
+import { Button, Slider } from '@osrd-project/ui-core';
+import type { Meta } from '@storybook/react';
+
+import '@osrd-project/ui-core/dist/theme.css';
+
+import { OPERATIONAL_POINTS, PATHS } from './lib/paths';
+import { PathLayer } from '../components/PathLayer';
+import { SpaceTimeChart } from '../components/SpaceTimeChart';
+import { type Point, type PathData, type OperationalPoint } from '../lib/types';
+import { getDiff } from '../utils/vectors';
+
+const DEFAULT_WIDTH = 1200;
+const DEFAULT_HEIGHT = 550;
+
+const MIN_ZOOM = 0;
+const MAX_ZOOM = 100;
+const MIN_ZOOM_MS_PER_PX = 600000;
+const MAX_ZOOM_MS_PER_PX = 625;
+const DEFAULT_ZOOM_MS_PER_PX = 7500;
+type SpaceTimeHorizontalZoomWrapperProps = {
+  offset: number;
+  operationalPoints: OperationalPoint[];
+  paths: (PathData & { color: string })[];
+};
+
+const zoomValueToTimeScale = (slider: number) =>
+  MIN_ZOOM_MS_PER_PX * Math.pow(MAX_ZOOM_MS_PER_PX / MIN_ZOOM_MS_PER_PX, slider / 100);
+
+const timeScaleToZoomValue = (timeScale: number) =>
+  (100 * Math.log(timeScale / MIN_ZOOM_MS_PER_PX)) /
+  Math.log(MAX_ZOOM_MS_PER_PX / MIN_ZOOM_MS_PER_PX);
+
+const SpaceTimeHorizontalZoomWrapper = ({
+  offset,
+  operationalPoints = [],
+  paths = [],
+}: SpaceTimeHorizontalZoomWrapperProps) => {
+  const [state, setState] = useState<{
+    zoomValue: number;
+    xOffset: number;
+    yOffset: number;
+    panning: null | { initialOffset: Point };
+  }>({
+    zoomValue: timeScaleToZoomValue(DEFAULT_ZOOM_MS_PER_PX),
+    xOffset: offset,
+    yOffset: 0,
+    panning: null,
+  });
+  useEffect(() => {
+    setState((prev) => ({ ...prev, xOffset: offset }));
+  }, [offset]);
+  const handleZoom = (zoomValue: number, position = DEFAULT_WIDTH / 2) => {
+    if (MIN_ZOOM <= zoomValue && zoomValue <= MAX_ZOOM) {
```

Comment:
Nit: would be slightly nicer to use `zoomValue = Math.min(MAX_ZOOM, Math.max(MIN_ZOOM, zoomValue))`, that way the zoom bounds are reached when using the mouse wheel  instead of being stuck on weird values close to the limit. (Or use `clamp()` from lodash.)

Reply:
done, I went for clamp